enterprise/frontend/src/metabase-enterprise/feature_level_permissions/index.ts

```diff
@@ -1,22 +1,18 @@
 import { hasPremiumFeature } from "metabase-enterprise/settings";
 import { PLUGIN_FEATURE_LEVEL_PERMISSIONS } from "metabase/plugins";
-import {
-  canAccessSettings,
-  canAccessDataModel,
-  canAccessDatabaseManagement,
-} from "./utils";
+import { canAccessDataModel, canAccessDatabaseManagement } from "./utils";
 
 import { getFeatureLevelDataPermissions } from "./permissions";
 import { DATA_COLUMNS } from "./constants";
 import {
   canDownloadResults,
   getDownloadWidgetMessageOverride,
 } from "./query-downloads";
+import { NAV_PERMISSION_GUARD } from "metabase/nav/utils";
 
 if (hasPremiumFeature("advanced_permissions")) {
-  PLUGIN_FEATURE_LEVEL_PERMISSIONS.canAccessSettings = canAccessSettings;
-  PLUGIN_FEATURE_LEVEL_PERMISSIONS.canAccessDataModel = canAccessDataModel;
-  PLUGIN_FEATURE_LEVEL_PERMISSIONS.canAccessDatabaseManagement = canAccessDatabaseManagement;
+  NAV_PERMISSION_GUARD["data-model"] = canAccessDataModel as any;
```

Comment:
You can avoid casting to `any` with something like this:
```
export interface UserFeaturePermissions {
  can_access_data_model: boolean;
  can_access_database_management: boolean;
}

export interface UserWithFeaturePermissions extends User {
  permissions?: UserFeaturePermissions;
}
```

You handle missing user anyway, so it would be just an extra `?.`, but you won't lose type-safety.